torchdata/datapipes/iter/transform/bucketbatcher.py

```diff
@@ -92,7 +103,104 @@ def __new__(
         if sort_key is not None:
             # In-batch shuffle each bucket seems not that useful, it seems misleading since .batch is called prior.
             if in_batch_shuffle:
-                datapipe = datapipe.batch(batch_size=bucket_num, drop_last=False).map(fn=_in_batch_shuffle_fn).unbatch()
+                datapipe = datapipe.batch(batch_size=bucket_num, drop_last=False).in_batch_shuffle().unbatch()
             else:
                 datapipe = datapipe.shuffle(buffer_size=bucket_size)
         return datapipe
+
+
+def _default_len_fn(token):
+    return len(token)
+
+
+def _token_len_fn(token, len_fn):
+    return len_fn(token), token
+
+
+def _token_min_filter_fn(data, *, min_len):
+    length, token = data
+    if length < min_len:
+        return False
+    return True
+
+
+def _token_max_filter_fn(data, *, max_len):
+    length, token = data
+    if length > max_len:
+        return False
+    return True
+
+
+@functional_datapipe("max_token_bucketize")
+class MaxTokenBucketizerIterDataPipe(IterDataPipe[DataChunk[T_co]]):
+    r"""
+    Iterable DataPipe to create mini-batches of data from bucket, and the
+    total length of each batch will be limited by ``max_token_count``.
+    (functional name: ``max_token_bucketize``).
+
```

Comment:
@nateanl or @ejguan can we add a one sentence description here about what the practical use case of this DataPipe will be?

We also did the same for the other DataPipe in this file because it seems complicated at a first glance and the use case isn't immediately obvious.